### DISCUSSION TITLE
ERR_CLEARTEXT_NOT_PERMITTED error for webview on android devices

### DISCUSSION THREAD

**Discussion by mehmury** [airsdk/Adobe-Runtime-Support#947] [Q&A]
I'm updating my apps on google play as aab. The latest air sdk is working. But in windows using webview in android apps, I see ERR_CLEARTEXT_NOT_PERMITTED error on 64 bit devices.
To solve this error, an additional xml is proposed that allows the domains I used.

<network-security-config>
    <domain-config cleartextTrafficPermitted="true">
        <domain includeSubdomains="true">your_domain1.com</domain>
        <domain includeSubdomains="true">your_domain2.com</domain>
    </domain-config>
</network-security-config>
I created this xml. I named it network_security_config.xml. I added this xml file to root directory in animate.
also in main xml
<application
    android:name=".MyApplication"
    android:networkSecurityConfig="@xml/network_security_config"......

should be defined as.
But after adding in this way, I am constantly getting an error while compiling. it doesn't recognize the xml i added.

Do you have any suggestions on solving this?

**Comment by ajwfrost** [comment]
Could you please zip up a small test case for this showing what you're doing, I can't really see why there would be a compile error for this...

thanks

**Reply by mehmury** [reply]
I am viewing web pages using webview feature in some of my applications. While testing on 64-bit devices in these applications, I got a page display error. I learned that an additional XML must be created for the allowed domains to solve this error.
I created this xml file and I wanted to compile it with animate by making the relevant additions, but I got an error.

I tried it with adt via cmd. As the last solution, I activated the KeepAndroidStudioOutput=true feature in adt.config and put the xml file I added in the temporary AndroidStudioProject file to res/xml/network_security_config.xml and compiled it again via cmd with adt. This time it was successful.
I guess when I add the xml file directly to the root folder, it can't put it under res/xml during compiling...

Thank you very much. I am attaching the screenshot and xml data as 
[zipcase.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/6759507/zipcase.zip)
a zipcase for the relevant addition.

**Reply by ajwfrost** [reply]
>  guess when I add the xml file directly to the root folder, it can't put it under res/xml during compiling...

Yes exactly: if you want to include this as a resource, you need to put it into a folder and then specify that this folder should be treated as containing resources.

So e.g. a folder in your application root called `network_res` under which you have an `xml` folder containing `network_security_config.xml`; then in your XML descriptor file, under the top-level application tag add:
```
<resdir>network_res</resdir>
```

It should also be possible to use a command-line argument of "-resdir" instead (it is for APK, but it's not yet working for AAB, it will be after the next release though!)

**Reply by mehmury** [reply]
Thaks..i put it manuel.and i solve This issue for now.NeXT update i try  New version of air sdk.
Good works.

**Reply by ajwfrost** [reply]
FYI the fix for the `-resdir` argument was in 33.1.1.554..
